source/slang/slang-emit-cuda.cpp

```diff
@@ -788,6 +788,29 @@ bool CUDASourceEmitter::tryEmitInstExprImpl(IRInst* inst, const EmitOpInfo& inOu
             m_writer->emit(")getOptiXRayPayloadPtr()");
             return true;
         }
+        case kIROp_GetOptiXHitAttribute:
+        {
+            auto typeToFetch = inst->getOperand(0);
+            auto idxInst = as<IRIntLit>(inst->getOperand(1));
+            IRIntegerValue idx = idxInst->getValue();
+            if (typeToFetch->getOp() == kIROp_FloatType) {
+                m_writer->emit("__int_as_float(optixGetAttribute_");
+            }
+            else
+            {
+                m_writer->emit("optixGetAttribute_");
+            }
+            m_writer->emit(idx);
```

Comment:
Nitpic - it might be slightly neater to have the special float logic all together rather (including the emit idx) instead of checking again for special casing the close.

Reply:
My guess is that eventually we'd want `GetOptiXHitAttribute` to always return a `uint`, and then have the code that inserts the new IR instructions also insert a bit-cast when it is needed. That would avoid mixing up the logic for fetching attributes and the logic for converting bits between types.